infinitered/react-native-mlkit
# Upgrade to Expo 51

There were a couple issues blocking this that I had to resolve:

1. The MLKit image labeling dependencies were pinned, when they shouldn't be
2. Expo changed the logger API so I had to replace `let logger = Logger()` with `let logger = Logger(logHandlers: [createOSLogHandler(category: Logger.EXPO_LOG_CATEGORY)])`
3. Used `expo-build-settings` to set the `minimumDeploymentTarget` and `compileSDKVersion` in `app.json`
4. Bumped expo version in `tsconfig` package
5. Deleted `yarn.lock` and resinstalled.
6. Replaced expo-modules scripts with custom scripts that check for Turbo. Messy but I'll circle back later to clean it up.
7. Unpinned Pods versions to allow Cocoapods to resolve the dependencies as needed

To test:
- Build and run the app. 
- Check if the various modules work as expected.
